lib/cdo/contact_rollups_validation.rb

```diff
@@ -0,0 +1,308 @@
+              WHERE hoc_organizer_years IS NOT NULL",
+      min: 375000,
+      max: 750000
+    },
+    {
+      name: "Count of contacts with grades taught",
+      query:  "SELECT COUNT(*) FROM contact_rollups_daily
+              WHERE grades_taught IS NOT NULL",
+      min: 200000,
+      max: 400000
+    },
+    {
+      name: "Check that all contacts with grades taught are "\
+            "teachers",
+      query: "SELECT COUNT(*) FROM contact_rollups_daily
+              WHERE grades_taught IS NOT NULL
+              AND Roles NOT LIKE '%Teacher%'",
+      min: 0,
+      max: 0
+    },
+    {
+      name: "Count of contacts with ages taught",
+      query:  "SELECT COUNT(*) FROM contact_rollups_daily
+              WHERE ages_taught IS NOT NULL",
+      min: 190000,
+      max: 400000
+    },
+    {
+      name: "Check that all contacts with ages taught are "\
+            "teachers",
+      query: "SELECT COUNT(*) FROM contact_rollups_daily
+              WHERE ages_taught IS NOT NULL
+              AND Roles NOT LIKE '%Teacher%'",
+      min: 0,
+      max: 0
+    },
+    {
+      name: "Count of contacts with forms submitted",
+      query:  "SELECT COUNT(*) FROM contact_rollups_daily
+              WHERE forms_submitted IS NOT NULL",
+      min: 2_600_000,
+      max: 5_000_000
+    },
+    {
+      name: "Check that all contacts with forms submitted are "\
+            "form submitters",
+      query: "SELECT COUNT(*) FROM contact_rollups_daily
+              WHERE forms_submitted IS NOT NULL
+              AND Roles NOT LIKE '%Form Submitter%'",
+      min: 0,
+      max: 0
+    },
+    {
+      name: "Count of contacts with form roles",
+      query:  "SELECT COUNT(*) FROM contact_rollups_daily
+              WHERE form_roles IS NOT NULL",
+      min: 1_750_000,
+      max: 3_500_000
+    },
+    {
+      name: "Check that all contacts with form roles are "\
+            "form submitters",
+      query: "SELECT COUNT(*) FROM contact_rollups_daily
+              WHERE form_roles IS NOT NULL
+              AND Roles NOT LIKE '%Form Submitter%'",
+      min: 0,
+      max: 0
+    },
+  ].freeze
+
+  def self.validate_contact_rollups
+    overall_pass = true
+
+    output = ""
+    # run each validation check
+    DATA_CHECKS.each do |check|
+      # run the validation query and get the returned count
+      count = PEGASUS_REPORTING_DB_READER[check[:query]].first.first[1]
+      # determine if the count is within validation bounds (inclusive)
+      pass = count >= check[:min] && count <= check[:max]
+
+      output_line = "#{check[:name]}: min #{check[:min]}, "\
+                    "max #{check[:max]}. Actual: #{count} -> #{pass_fail_string(pass)}"
+      log output_line
+
+      output += output_line + '\n'
+
+      # keep track of if we have an overall pass
+      overall_pass = false unless pass
+    end
+
+    output += "Overall result: #{pass_fail_string(overall_pass)}"
+
+    return overall_pass, output
```

Comment:
It might be worth making it explicit that you are returning an array, e.g., `return [overall_pass, output]`. If you do, the `return` becomes unnecessary.